src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb

```diff
@@ -145,6 +146,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             Return trivia.IsKind(SyntaxKind.SkippedTokensTrivia)
         End Function
 
+        <Extension()>
+        Public Function IsGlobalStatementContext(syntaxTree As SyntaxTree, position As Integer, cancellationToken As CancellationToken) As Boolean
+            If Not syntaxTree.IsInteractiveOrScript() Then
+                Return False
+            End If
+
+            Dim token As SyntaxToken = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken).GetPreviousTokenIfTouchingWord(position)
+
+            If token.IsKind(SyntaxKind.None) Then
```

Comment:
What case does this address?  It looks like it might be trying to cover BOF, but I'm not sure.